src/v/config/configuration.cc

```diff
@@ -829,6 +829,25 @@ configuration::configuration()
       "required. see also `kafka_enable_authorization`",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
       false)
+  , sasl_mechanisms(
+      *this,
+      "sasl_mechanisms",
+      "A list of supported SASL mechanisms. `SCRAM` is allowed.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      {"SCRAM"},
+      validate_sasl_mechanisms)
+  , sasl_kerberos_keytab(
+      *this,
+      "sasl_kerberos_keytab",
+      "The location of the Kerberos keytab file for Redpanda",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      "/var/lib/redpanda/redpanda.keytab")
+  , sasl_kerberos_principal(
```

Comment:
Since per-node config is not desirable, I've made it work this way:

The SPN is made if 3 parts, `<primary>/<host>@<realm>`.

The primary should be specified here.

We're using `gss_import_name` with `GSS_C_NT_HOSTBASED_SERVICE`. 

So if the KDC has an entry for each host, and that entry is requested for the keytab, then no node-level configuration is required in Redpanda, it should be found during `gss_acquire_cred_from`.

This is how the test does it.

Reply:
:+1: